modules/basics/src/main/java/com/opengamma/basics/currency/FxMatrix.java

```diff
@@ -0,0 +1,557 @@
+/**
+ * Copyright (C) 2011 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.basics.currency;
+
+import static com.opengamma.collect.Guavate.entriesToImmutableMap;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.collect.ArgChecker;
+import com.opengamma.collect.tuple.Pair;
+
+/**
+ * Immutable class describing a set of currencies and all the cross rates between them.
+ */
+public class FxMatrix {
```

Comment:
You are probably correct, but as we are unsure exactly what, if any, role this class will play I'm not going to address for the time being.